src/wallet/wallet.cpp

```diff
@@ -3004,6 +3004,20 @@ std::map<std::pair<CTxDestination, Optional<CTxDestination>>, std::vector<COutpu
     return result;
 }
 
+std::map<libzcash::SaplingPaymentAddress, std::vector<SaplingNoteEntry>> CWallet::ListNotes() const
+{
+    std::map<libzcash::SaplingPaymentAddress, std::vector<SaplingNoteEntry>> result;
+
+    std::vector<SaplingNoteEntry> notes;
+    Optional<libzcash::SaplingPaymentAddress> dummy = nullopt;
+    m_sspk_man->GetFilteredNotes(notes, dummy);
+    for (const auto& note : notes) {
+        result[note.address].emplace_back(std::move(note));
+    }
+
+    return result;
+}
+
```

Comment:
nit: would be better to place this inside the sspkm.

Reply:
Moved it in the SaplingScriptPubKeyMan and made a wrapper in the wallet.